presto-verifier/src/main/java/com/facebook/presto/verifier/QueryRewriter.java

```diff
@@ -147,8 +148,26 @@ private String createTemporaryTableName()
             throws SQLException
     {
         com.facebook.presto.sql.tree.Query createSelectClause = createTableAsSelect.getQuery();
+
         // Rewrite the query to select zero rows, so that we can get the column names and types
-        com.facebook.presto.sql.tree.Query zeroRowsQuery = new com.facebook.presto.sql.tree.Query(createSelectClause.getWith(), createSelectClause.getQueryBody(), ImmutableList.of(), Optional.of("0"), createSelectClause.getApproximate());
+        QueryBody innerQuery = createSelectClause.getQueryBody();
+        com.facebook.presto.sql.tree.Query zeroRowsQuery;
+        if (innerQuery instanceof QuerySpecification) {
+            QuerySpecification querySpecification = (QuerySpecification) innerQuery;
+            innerQuery = new QuerySpecification(
+                    querySpecification.getSelect(),
+                    querySpecification.getFrom(),
+                    querySpecification.getWhere(),
+                    querySpecification.getGroupBy(),
+                    querySpecification.getHaving(),
+                    querySpecification.getOrderBy(),
+                    Optional.of("0"));
```

Comment:
Is there a reason to put the `0` here instead of on the `Query`? If not, I would put it on the `Query` and make this one `empty()` just to make it consistent